File: PDOO-Civitas/CivitasR/lib/sorpresa_salvoconducto.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Sorpresa_salvoconducto < Sorpresa
    public
    def initialize(mazo)
      
      @mazo = mazo
      @texto = "Esta sorpresa evita que caigas en la cárcel"
      
    end
    
    public
    def aplicar_a_jugador(actual, todos)
      
      if(jugador_correcto(actual, todos))
        
        informe(actual, todos)
        i = 0
        tiene = false
                    
        while(i<todos.length())
                
          tiene = todos.at(i).tiene_salvoconducto
          i=i+1
        end
            
        if(!tiene)
            
          todos.at(actual).obtener_salvoconducto(self)
          salir_del_mazo()
        end
      end
      
    end
    
    private
    def informe(actual, todos)
      
      Diario.instance.ocurre_evento("Se esta aplicando una sorpresa SALIR DE LA CÁRCEL al jugador " + todos.at(actual).to_s);
      
    end
    
    public
    def to_s()
      
      return "Tipo: Sorpresa Salvoconducto\n" + @texto

    end

    public_class_method :new
    
  end
end

File: PDOO-Civitas/CivitasJ/src/civitas/SorpresaSalvoconducto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ramonjtorres
 */
public class SorpresaSalvoconducto extends Sorpresa{
    
    private String texto;
    MazoSorpresas mazo;
    
    SorpresaSalvoconducto(MazoSorpresas mazo){
    
        this.mazo = mazo;
        this.texto = "Esta sorpresa evita que caigas en la cárcel";
        this.informacion = this.texto;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
    
        if(this.jugadorCorrecto(actual, todos)){
        
            this.informe(actual, todos);
            int i = 0;
            boolean tiene = false;
                    
            while(i<todos.size()){
                
                tiene = todos.get(i).tieneSalvoconducto();
                i++;
            }
            
            if(!tiene){
            
                todos.get(actual).obtenerSalvoconducto(this);
                this.salirDelMazo();
            }
        }
        
    }
    
    @Override
    void informe(int actual, ArrayList<Jugador> todos){
    
        Diario.getInstance().ocurreEvento("Se esta aplicando una sorpresa SALIR DE LA CÁRCEL al jugador " + todos.get(actual).getNombre());
    }
    @Override
    public String toString(){
        
        String sorpresa = "Tipo: Sorpresa Salvoconducto\n";
        
        sorpresa += this.texto;
        
        return sorpresa;
    }
}
